File: src/lib/software/minecraft_launcher_legacy.py
from src.lib.software import BaseSoftware, SoftwareCategory

class MinecraftLauncherLegacy(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'minecraft-launcher-legacy'
        self.name = 'Minecraft Launcher (Legacy)'
        self.category = SoftwareCategory.Gaming
        self.download_name = 'MinecraftInstaller.msi'
        self.is_archive = False
        self.should_cache_url = False
        self.requires_admin = False
        self.icon = 'minecraft-launcher-legacy.png'
        self.homepage = 'https://minecraft.net'

    def resolve_download_url(self):
        self.url_resolved.emit('https://launcher.mojang.com/download/MinecraftInstaller.msi?ref=mcnet')

File: packages/app/src/services/software/definitions/minecraftLauncherLegacy.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'minecraft-launcher-legacy';
	public name = 'Minecraft Launcher (Legacy)';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'MinecraftInstaller.msi';
	public icon = 'minecraft-launcher-legacy.png';
	public homepage = 'https://minecraft.net';

	public async resolveDownloadUrl() {
		return ok('https://launcher.mojang.com/download/MinecraftInstaller.msi?ref=mcnet');
	}
}
